### PR TITLE
Fixed: #16810, inverted images.

### PR DESCRIPTION
**Pull request by DJTechnoo**
Fixed: #16810, inverted image pattern on maps.

### CODE REVIEW

**Comment by TorsteinHonsi** [comment]
@DJTechnoo It looks like in the current state of the PR, the scaling is done on the pattern in general. But we already have a hook, `Point.calculatePatternDimensions`, for adapting the pattern to individual points. Maybe we should adapt that to not only work with width and height, but also the scale transform?

At least in theory, that would be more correct because it won't flip the pattern unless it is _actually_ applied to a flipped point. With the current approach it flips the pattern regardless of where it is used.

**Comment by DJTechnoo** [comment]
Thank you, I'll dive into it right away!

File: samples/unit-tests/accessibility/accessible-highmaps/demo.js [hunk 1]
```diff
@@ -1,13 +1,16 @@
 QUnit.test('Basic map', function (assert) {
-    var chart = Highcharts.mapChart('container', {
+    const polishFlagURL =
```
Comment (raf18seb) [inline samples/unit-tests/accessibility/accessible-highmaps/demo.js:2]: The variable name without indicating a value would be better, e.g.
```js
const url = ...
```

File: samples/unit-tests/accessibility/accessible-highmaps/demo.js [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
+        'https://upload.wikimedia.org/wikipedia/en/1/12/Flag_of_Poland.svg';
+    const chart = Highcharts.mapChart('container', {
             chart: {
                 map: 'custom/europe'
             },
             series: [
                 {
+                    keys: ['hc-key', 'color.pattern.image'],
```
Comment (raf18seb) [inline samples/unit-tests/accessibility/accessible-highmaps/demo.js:10]: You have 2 keys defined, but your data point has 3 values: `['pl', polishFlagURL, 2],`

File: ts/Extensions/PatternFill.ts [hunk 3]
```diff
@@ -223,6 +224,15 @@ Point.prototype.calculatePatternDimensions = function (
         if (!bBox.width || !bBox.height) {
             pattern._width = 'defer';
             pattern._height = 'defer';
+
+            // Check if map image pattern is flipped (#16810)
```
Comment (raf18seb) [inline ts/Extensions/PatternFill.ts:228]: This comment here is repeated in 2 different places but both parts of code are different so I think one of the comments should be changed or should be more descriptive

File: ts/Extensions/PatternFill.ts [hunk 4]
```diff
@@ -341,10 +351,20 @@ SVGRenderer.prototype.addPattern = function (
...
+        if (options.patternTransform) {
+            options.patternTransform += ' scale(1, -1)';
+        }
+    }
+
     if (options.patternTransform) {
         attrs.patternTransform = options.patternTransform;
     }
 
+
```
Comment (raf18seb) [inline ts/Extensions/PatternFill.ts:367]: Unnecessary line break methinks🤔

**Review by raf18seb — CHANGES_REQUESTED** [review]
Just a few small things, see the comments below

**Review by KacperMadej — APPROVED** [review]
What Rafał wrote. Other than that, looks good to me 👍

File: samples/unit-tests/accessibility/accessible-highmaps/demo.js [hunk 5]
```diff
@@ -23,6 +26,13 @@ QUnit.test('Basic map', function (assert) {
         description.indexOf('Europe') > 0,
         'There is a screen reader region, and it contains "Europe"'
     );
+
+    assert.strictEqual(
+        chart.series[0].transformGroups[0].scaleY < 0,
+        // eslint-disable-next-line no-underscore-dangle
+        chart.series[0].points[1].options.color.pattern._inverted,
```
[the remark this answers is not in the document]
Comment (DJTechnoo) [reply samples/unit-tests/accessibility/accessible-highmaps/demo.js:33]: After trying this, ESLint wanted me to use the dotted notation again  ಥ_ಥ

**Review by raf18seb — APPROVED** [review]
Thanks!

**Review by TorsteinHonsi — APPROVED** [review]
Awesome, thanks!

**Comment by TorsteinHonsi** [comment]
Now pending CI run after merging master in.

**Pull request merged**